### ISSUE TITLE
About backward

### ISSUE THREAD

**Issue by lk1048453160** [HanxunHuangLemonBear/SCELoss-Reproduce#3]
Hi, I want to know which file is backpropagation in, about SCELoss. thanks!

**Comment by HanxunH** [comment]
I'm not sure what do you mean?
The loss function is in https://github.com/HanxunH/SCELoss-Reproduce/blob/be22ef6488b5a37e9b67cafbeb846b7385d502c4/loss.py#L5

**Comment by lk1048453160** [comment]
> I'm not sure what do you mean?
> The loss function is in
> 
> https://github.com/HanxunH/SCELoss-Reproduce/blob/be22ef6488b5a37e9b67cafbeb846b7385d502c4/loss.py#L5

Hi, thanks very much! I'm a little confused, because when I read some code in the pytorch environment, every loss function has a backward function, but all of them are official loss function. SCELoss is  custom, so I want to know SCELoss.backward().
Thanks very much!

**Comment by lk1048453160** [comment]
> I'm not sure what do you mean?
> The loss function is in
> 
> https://github.com/HanxunH/SCELoss-Reproduce/blob/be22ef6488b5a37e9b67cafbeb846b7385d502c4/loss.py#L5

I find forward, not backward. just like this. thanks for your reply!
![image](https://user-images.githubusercontent.com/54702384/94506882-054d1680-0241-11eb-9c3a-b23a5c0df1a6.png)

**Comment by HanxunH** [comment]
All the function used in forward() has backward() already implemented by PyTorch. So, there is no need to write custom backward() in this case.
i.e. F.softmax(). torch.log() torch.sum() 

Does this help?

**Comment by lk1048453160** [comment]
> All the function used in forward() has backward() already implemented by PyTorch. So, there is no need to write custom backward() in this case.
> i.e. F.softmax(). torch.log() torch.sum()
> 
> Does this help?

yes ,thanks very much. But I think this loss function SEC, is user-defined. There should be no corresponding back propagation in pytorch.  So I think we should set up our own .backward()
loss = SECLoss(out, label)
loss.backward()
Just like this.
I'm a little confused. I hope I can get your help!
Thanks again!

**Comment by HanxunH** [comment]
> > All the function used in forward() has backward() already implemented by PyTorch. So, there is no need to write custom backward() in this case.
> > i.e. F.softmax(). torch.log() torch.sum()
> > Does this help?
> 
> yes ,thanks very much. But I think this loss function SEC, is user-defined. There should be no corresponding back propagation in pytorch. So I think we should set up our own .backward()
> loss = SECLoss(out, label)
> loss.backward()
> Just like this.
> I'm a little confused. I hope I can get your help!
> Thanks again!

You do not need .backward() since it inherits from torch.nn.Module
